src/net_processing.cpp

```diff
@@ -2004,6 +2004,13 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 instantsend.AcceptLockRequest(txLockRequest);
                 instantsend.Vote(tx.GetHash(), connman);
             }
+            else
+            {
+                if(CInstantSend::IsTrxSimple(tx) && CInstantSend::CanAutoLock()) {
+                    if(!instantsend.ProcessTxLockRequest(tx, connman))
```

Comment:
Thinking about this piece of code a bit more and... it probably won't do anything actually. Instead, I think you should process ix and simple txes identically. One way would be to change `(strCommand == NetMsgType::TXLOCKREQUEST)` conditions into `(strCommand == NetMsgType::TXLOCKREQUEST || (strCommand == NetMsgType::TX && CInstantSend::IsTrxSimple(tx) && CInstantSend::CanAutoLock()))` here (lines 2001+), a bit above (lines 1949+) and for rejected txes down below (on lines 2131+). This would also require smth like `txLockRequest.tx = ptx;` on lines 1931+.

However, I'm not 100% sure if thee changes would be enough either. We should probably adjust `p2p-instantsend.py` to include some tests for simple txes (success for simple txes, failure for non-simple regular txes) to be a bit more confident that changes in this PR actually work.

Reply:
Agree,  adding tests now.